pennylane/fourier/spectrum.py

```diff
@@ -48,7 +48,8 @@ def stop_at(obj):
         """
 
         if isinstance(obj, qml.operation.Operation):
-            takes_input = any(p in original_inputs for p in obj.parameters)
+            trainable = tape.get_parameters(trainable_only=True)
+            takes_input = any(p in trainable for p in obj.parameters)
```

Comment:
That was the wrong logic previously...if the operation already processed a trainable parameter, then `takes_input` would be false, even if it does take an input and should raise an error!